### DISCUSSION TITLE
ComfortSync A3 (Armstrong Air) vs Lennox iComfort S30

### DISCUSSION THREAD

**Discussion by IonCaza** [PeteRager/lennoxs30#266] [General]
Hey there, these devices seem identical. I'm assuming the firmware on them is slightly different to connect to different APIs/cloud providers. I have a ComfortSync A3. I'm wondering if I can leverage your implementation and just change the hostname of the API to point to ComfortSyncs? Maybe someone has already tried this.

Can you tell me where you're defining the cloud API hostname in the code?

**Comment by PeteRager** [comment]
Those URLs are defined in the API

https://github.com/PeteRager/lennoxs30api/blob/19549dea5ed0ac73679b645bb774bf8f6b57905e/lennoxs30api/s30api_async.py#L70

 The first thing I'd try is to see if the local connection works,  using the IP address of the thermostat. It connects on port 443.

If local does not work then we'll need to figure out what endpoints the app is talking to.  Easiest way to do that is to run it through a burp proxy.

**Reply by IonCaza** [reply]
Local does work! I don't really need it connecting to the cloud since I wanted this to work locally to begin with. Thank you so much!

**Comment by PeteRager** [comment]
That's great.  I'll add this to the compatible list.  If you could also grab a couple of screen shots so I can understand what this thing is that would be helpful.

Go to integrations / and click on the devices link for the Lennox integration and Take a screen shot of the devices page

And for the device that calls itself an S30 go into and grab a screen shot of that device details page.

**Reply by IonCaza** [reply]
I attached some screenshots below. Here's a link to the product: https://www.armstrongair.com/products/comfort-sync-thermostat It's literally a rebranded S30...

<img alt="image" src="https://github.com/PeteRager/lennoxs30/assets/1081534/ee6e76be-e529-4de5-b8eb-8e8838864856">
<img alt="image" src="https://github.com/PeteRager/lennoxs30/assets/1081534/1e1c88b6-8d6c-4af8-99dd-4d5741f90fe8">
<img alt="image" src="https://github.com/PeteRager/lennoxs30/assets/1081534/d8915c60-036f-4ffe-8cc0-9d7e263b9474">

**Comment by cap60552** [comment]
I can also confirm that this works with the ComfortSync A3.   THANK YOU!!!

**Reply by PeteRager** [reply]
Thanks for the note, I've updated the product docs to add this in.  You are using a local connection? Any details on getting it setup?

**Reply by cap60552** [reply]
It was very painless to get setup to be honest.     

1. Assign a static IP to the "thermostat" in my router using DHCP. 
2. Install the lennoxs30 integration in HA using HACS, restart HA.
3. On the integrations screen, add the Lennox S30/E30 integration.
4. During the setup workflow I entered the IP of the thermostat, as set in step 1.
5. I also checked the options for "Temperature and humidity Sensors","Create allergenDefender switch","Diagnostic Sensors"
6. After config was complete, I went in to the entity list, and set the diagnostic level to 1, to allow the diagnostic sensors to collect data.

After that, it was a matter of adding the various sensors, and the climate control to my various dashboards, and tabs, and then updating automations to use the new entities.
